### PR TITLE
[Darwin] Fix availability of exp10 for watchOS, tvOS, xROS.

### PR DESCRIPTION
**Pull request by fhahn**
Update availability information added in 1eb7f055d9a. exp10 is available on iOS >= 7.0 and macOS >= 10.9. On all other platforms, it is available on any version. Also drop the x86 check, as the availability only depends on the OS version, not the target platform.

### CODE REVIEW

File: llvm/lib/CodeGen/TargetLoweringBase.cpp [hunk 1]
```diff
@@ -240,18 +240,17 @@ void TargetLoweringBase::InitLibcalls(const Triple &TT) {
...
+      [[fallthrough]];
     case Triple::TvOS:
     case Triple::WatchOS:
     case Triple::XROS:
-      if (!TT.isWatchOS() &&
-          (TT.isOSVersionLT(7, 0) || (TT.isOSVersionLT(9, 0) && TT.isX86()))) {
-        setLibcallName(RTLIB::EXP10_F32, nullptr);
-        setLibcallName(RTLIB::EXP10_F64, nullptr);
-      } else {
         setLibcallName(RTLIB::EXP10_F32, "__exp10f");
```
Comment (cachemeifyoucan) [inline llvm/lib/CodeGen/TargetLoweringBase.cpp:252]: Format?

**Review by cachemeifyoucan — APPROVED** [review]
LGTM

**Review by porglezomp — APPROVED** [review]
LGTM. Commit message suggestion while you're fixing formatting, I think it should say:
```diff
-On all platforms
+On all other platforms
```

File: llvm/lib/CodeGen/TargetLoweringBase.cpp (hunk 1 above)
Comment (fhahn) [reply llvm/lib/CodeGen/TargetLoweringBase.cpp]: Should be fixed now! I guess my clang-format-diff setup may miss some of those cases....

**Comment by fhahn** [comment]
> LGTM. Commit message suggestion while you're fixing formatting, I think it should say:
> 
> ```diff
> -On all platforms
> +On all other platforms
> ```

Thanks, should be fixed

**Pull request merged**